From 71f572540af739d4d22ca4eecca3c333a1726cb3 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Mon, 10 Sep 2018 11:12:12 +0200
Subject: [PATCH 021/188] arm: dt: imx6qdl-mba6: [HACK]: add spidev nodes

needs special patch for spidev to have a generic comptible.

Attention: not for mainline

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 arch/arm/boot/dts/imx6qdl-mba6.dtsi | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm/boot/dts/imx6qdl-mba6.dtsi b/arch/arm/boot/dts/imx6qdl-mba6.dtsi
index e9478fb22905..6d6c3fffc15c 100644
--- a/arch/arm/boot/dts/imx6qdl-mba6.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-mba6.dtsi
@@ -172,6 +172,20 @@
 	status = "okay";
 };
 
+&ecspi1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_ecspi1>, <&pinctrl_ecspi1_mba6>;
+	fsl,spi-num-chipselects = <2>;
+	cs-gpios = <&gpio3 19 0>, <&gpio3 24 0>;
+
+	spidev1_1: spidev1-1@1 {
+		compatible = "tq,testdev";
+		spi-max-frequency = <1000000>;
+		reg = <1>;
+		status = "okay";
+	};
+};
+
 &fec {
 	phy-mode = "rgmii-id";
 	phy-reset-gpios = <&gpio1 25 GPIO_ACTIVE_LOW>;
@@ -245,6 +259,12 @@
 			>;
 		};
 
+		pinctrl_ecspi1_mba6: ecspimba6grp {
+			fsl,pins = <
+				MX6QDL_PAD_EIM_D24__GPIO3_IO24 0xb099 /* eCSPI1 SS2 */
+			>;
+		};
+
 		pinctrl_enet: enetgrp {
 			fsl,pins = <
 				/* FEC phy IRQ */
